probe/process/walker.go

```diff
@@ -20,13 +20,14 @@ type Walker interface {
 	Walk(func(Process, Process)) error
 }
 
-// CachingWalker is a walker than caches a copy of the output from another
-// Walker, and then allows other concurrent readers to Walk that copy.
+// CachingWalker is a walker that wraps a platform-specific walker (source),
+// triggers it at every Tick() and caches a copy of the output, to allow
+// other concurrent readers to Walk that copy.
```

Comment:
The changes in this file belong to another PR: they are unrelated to the ebpf work.